app/Http/Controllers/TalksController.php

```diff
@@ -157,10 +141,72 @@ public function show($id)
 
     public function destroy($id)
     {
-        Auth::user()->talks()->findOrFail($id)->delete();
+        Auth::user()->talks()->withTrashed()->findOrFail($id)->forceDelete();
 
         Session::flash('message', 'Successfully deleted talk.');
 
         return Redirect::to('talks');
     }
+
+    public function softDelete($id)
+    {
+        Auth::user()->talks()->findOrFail($id)->delete();
+
+        Session::flash('message', 'Successfully archived talk.');
+
+        return Redirect::to('talks');
+    }
+
+    public function restore($id)
+    {
+        Auth::user()->talks()->withTrashed()->findOrFail($id)->restore();
+
+        Session::flash('message', 'Successfully restored talk.');
+
+        return Redirect::to('archive');
+    }
+
+
+    public function archiveIndex()
+    {
+        list($talks, $sorting_talk) = $this->sortTalks('archive');
+
+        return View::make('talks.archive')
+          ->with('talks', $talks)
+          ->with('sorting_talk', $sorting_talk);
+    }
+
+  /**
+   * Sorts through talks and returns an array of variables. Access the variables using the list() construct.
+   * @param  string $type Whether the index is for archived talks or active talks
+   * @return array       Returns array of variables.
+   */
+    private function sortTalks($type = 'active')
+    {
+        $bold_style = 'style="font-weight: bold;"';
+
+        $sorting_talk = $this->sorting_talks;
+
+        if ($type == 'archive') {
```

Comment:
I think I'd just pass the `$talks` variable into `sortTalks` instead of passing the type.